src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -460,27 +448,12 @@ private async ValueTask<ProtocolToken> ReceiveBlobAsync<TIOAdapter>(TIOAdapter a
                  where TIOAdapter : IReadWriteAdapter
         {
             await FillHandshakeBufferAsync(adapter, SecureChannel.ReadHeaderSize).ConfigureAwait(false);
-            if (_framing == Framing.Unified || _framing == Framing.Unknown)
-            {
-                _framing = DetectFraming(_handshakeBuffer.ActiveReadOnlySpan);
-            }
-
-            if (_framing != Framing.SinceSSL3)
-            {
-#pragma warning disable 0618
-                _lastFrame.Header.Version = SslProtocols.Ssl2;
-#pragma warning restore 0618
-                _lastFrame.Header.Length = GetFrameSize(_handshakeBuffer.ActiveReadOnlySpan) - TlsFrameHelper.HeaderSize;
-            }
-            else
-            {
-                TlsFrameHelper.TryGetFrameHeader(_handshakeBuffer.ActiveReadOnlySpan, ref _lastFrame.Header);
-            }
+            TlsFrameHelper.TryGetFrameHeader(_handshakeBuffer.ActiveReadOnlySpan, ref _lastFrame.Header);
 
             if (_lastFrame.Header.Length < 0)
             {
                 if (NetEventSource.Log.IsEnabled()) NetEventSource.Error(this, "invalid TLS frame size");
-                throw new IOException(SR.net_frame_read_size);
+                throw new AuthenticationException(SR.net_frame_read_size);
```

Comment:
Why to change this exception when you leave IOException in GetFrameSize?
In other words: What is the rule to throw IOException vs. AuthenticationException?

Reply:
ServerAsyncAuthenticate_InvalidHello_Throws checks for the exception type. In the past we can get one or the other depending on connection management. So I had to choose. Since getting invalid length did not feel like IO problem, I decided to use `AuthenticationException`